### DISCUSSION TITLE
How to use GitHub webhook with WorkflowEventBinding? Webhook doesn't support "Authorization" header

### DISCUSSION THREAD

**Discussion by Sudhamsh** [argoproj/argo-workflows#7975] [Q&A]
I'm trying to trigger a workflow using WorkflowEventBinding and GitHub webhook. I followed below example, curl works fine. But I'm not able to trigger it via GitHub webhook as I can't configure the "Authorization" header with my Bearer token. Am I missing something, or is there a way to authorize with a secret ?

https://argoproj.github.io/argo-workflows/events/#workflow-template-triggered-by-the-event

**Comment by davidedmondsMPG** [comment]
This showed up on a google search while trying to get a webhook set up - there is support for GitHub webhooks and other similar tooling as of at least v3.5.5 - see https://argo-workflows.readthedocs.io/en/stable/webhooks/ for an overview. It is helpful to follow the argo-server's logs when webhooks are being sent to debug what issues exist - Argo expects particular names for the service account and service account token secret.

**Comment by wesleyscholl** [comment]
@davidedmondsMPG @Sudhamsh 

I followed this guide and was unable to get the GitHub webhook and WorkflowEventBinding to function as intended. (I could locally trigger the WorkflowEventBinding without issue) I also setup argo events and was able to trigger Argo Workflows from GitHub webhook push events. I think the WorkflowEventBinding  GitHub integration would be very useful to the community. @alexec This area of the docs could use improvement.

https://argo-workflows.readthedocs.io/en/stable/webhooks/
